lib/io.toit

```diff
@@ -1199,22 +1255,15 @@ class Buffer extends CloseableWriter:
     buffer.little-endian.write-int32 0x12345678
     // The least significant byte 0x78 is at index 0.
     print buffer.bytes  // => #[0x78, 0x56, 0x34, 0x12]
-  ```
 
-  ```
-  import io
-
-  main:
-    buffer := io.Buffer
-    writer := buffer.little-endian
-    writer.write "Can be used like a normal writer."
-    writer.write-int32 0x12345678
-    result := buffer.bytes
-    ...
+    // The buffer version also supports 'put' operations
+    buffer.little-endian.put-int32 --at=0 0x11223344
   ```
   */
   little-endian -> EndianBuffer:
-    return EndianBuffer --buffer=this --byte-order=binary.LITTLE_ENDIAN
+    if not endian_ or endian_.byte-order_ != binary.LITTLE_ENDIAN:
```

Comment:
I wouldn't read the field that many times.

Reply:
done.